templates/components/multiagent/typescript/workflow/tools.ts

```diff
@@ -0,0 +1,320 @@
+import { Context } from "@llamaindex/core/workflow";
+import fs from "fs/promises";
+import {
+  BaseToolWithCall,
+  callTool,
+  ChatMessage,
+  ChatResponse,
+  ChatResponseChunk,
+  PartialToolCall,
+  QueryEngineTool,
+  ToolCall,
+  ToolCallLLM,
+  ToolCallLLMMessageOptions,
+} from "llamaindex";
+import crypto from "node:crypto";
+import path from "path";
+import { getDataSource } from "../engine";
+import { createTools } from "../engine/tools/index";
+import { AgentRunEvent } from "./type";
+
+export const getQueryEngineTool = async (
```

Comment:
TBD: Can add advanced feature from LlamaCloud index, but it'd make the code more complicated.